contracts/Contribution.sol

```diff
@@ -18,14 +18,21 @@ contract Contribution {
     tokenAddress = Token(_tokenAddress);
   }
 
+  mapping(address => uint256) public donations;
+
   function sendContribution() public payable {
     require(msg.value > 0, "Amount must not be zero.");
+    donations[msg.sender] += msg.value;
```

Comment:
Always avoid using operations in smart contract. 
`donations[msg.sender] = donations[msg.sender].add(msg.value)` is better

Reply:
Also. You want to update state _after_ any functions that could potentially revert. Here, `_sendTokens` could revert, but you've already updated the donations mapping.